src/finance/loan.cto

```diff
@@ -0,0 +1,35 @@
+namespace org.accordproject.finance.loan
+
+import org.accordproject.time.Duration from https://models.accordproject.org/time.cto
+import org.accordproject.money.* from https://models.accordproject.org/money.cto
+
+enum CurrencyType {
+  o MONETARY
+  o CRYPTOMONETARY
+}
+
+concept PaymentFrequency {
+    o Integer periodMultiplier
+    o Duration period
+}
+
+/**
+ * Definition of a Loan, based on the FpML schema:
+ * http://www.fpml.org/spec/fpml-5-3-2-wd-2/html/reporting/schemaDocumentation/schemas/fpml-asset-5-3_xsd/complexTypes/Loan.html
+ *
+ */
+concept Loan {
+    o String[] instrumentId
+    o String description optional
+    o CurrencyType currencyType
+    o CurrencyCode currencyCode optional
+    o CryptoCurrencyCode cryptoCurrencyCode optional
```

Comment:
Instead of two optional currency codes, you could create a type hierarchy that captures the variance. 
For example, a parent type such as `FiatOrDigitalAmount` (with a better name) in https://models.accordproject.org/money that `MonetaryAmount` and `CryptoMonetaryAmount` both extend.

Your model could then include a single attribute of type `FiatOrDigitalAmount`  that captures the value and currency code.